src/util/mime.js

```diff
@@ -16,22 +29,21 @@ export function detectContentType (fileName, bytes) {
     if (info?.mime) return info.mime
     if (infos[0].mime) return infos[0].mime
   }
-  return lookup(fileName) || detectTextContent(bytes)
 }
 
 /** @param {Uint8Array} bytes */
-function detectTextContent (bytes) {
+function detectTextContent (bytes, defaultMime = 'text/plain') {
   const encoding = chardet.detect(bytes)
   if (!encoding) return
-  let mime = 'text/plain'
+  let mime = defaultMime
   if (encoding === 'UTF-8' || encoding === 'ISO-8859-1') {
     const text = toString(bytes).toLowerCase()
     if (text.startsWith('<!doctype html')) {
       mime = 'text/html'
     } else if (text.includes('<svg')) {
       mime = 'image/svg+xml'
-    } else if (text.startsWith('<!xml')) {
-      mime = 'text/xml'
+    } else if (text.startsWith('<?xml')) {
```

Comment:
! is for doctypes, ? is for declarations.
```xml
<?xml version="1.0"?>
<!DOCTYPE greeting SYSTEM "hello.dtd">
<greeting>Hello, world!</greeting> 
```
– https://www.w3.org/TR/REC-xml/#sec-prolog-dtd

Reply:
which is odd if you think about it. A question is not a statement. A question mark for a _"declaration"_ ~~sucks~~ is counter intuitive.